quokka/core/views.py

```diff
@@ -555,3 +555,44 @@ def get(self, tag):
         )
 
         return self.make_rss(feed_name, contents)
+
+
+class SiteMap(MethodView):
+    @staticmethod
+    def make_external_url(url):
+        return urljoin(request.url_root, url)
+
+    def get_contents(self):
+        now = datetime.now()
+        filters = {
+            'published': True,
+            'available_at__lte': now,
+        }
+
+        contents = Content.objects().filter(**filters)
+        return contents
+
+    def sitemap_render(self, contents):
+        tmpl = """
```

Comment:
:+1: 

What about using a Jinja template to render it?

Also we need to include channels and all the fixed urls.

Lets work more on this branch :)